api/v1alpha1/datadogagent_types.go

```diff
@@ -14,12 +14,23 @@ import (
 	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
+// DatadogFeatures are features which are running on the agent as well on the clusterAgent.
+// +k8s:openapi-gen=true
+type DatadogFeatures struct {
+	// OrchestratorExplorer configuration
+	OrchestratorExplorer *OrchestratorExplorerConfig `json:"orchestratorExplorer,omitempty"`
+}
+
 // DatadogAgentSpec defines the desired state of DatadogAgent
 // +k8s:openapi-gen=true
 type DatadogAgentSpec struct {
 	// Configure the credentials required to run Agents
 	Credentials AgentCredentials `json:"credentials"`
 
+	// DatadogFeatures are features which are running on the agent as well on the clusterAgent.
+	// +optional
+	DatadogFeatures *DatadogFeatures `json:"datadogFeatures,omitempty"`
```

Comment:
I think `features` for the json name is enough. we already know that we are configuring the datadog agent :)

 ```suggestion
	   DatadogFeatures *DatadogFeatures `json:"features,omitempty"`
```